src/NzbDrone.Core/RootFolders/RootFolderService.cs

```diff
@@ -153,7 +154,34 @@ private List<UnmappedFolder> GetUnmappedFolders(string path)
             foreach (string unmappedFolder in unmappedFolders)
             {
                 var di = new DirectoryInfo(unmappedFolder.Normalize());
-                results.Add(new UnmappedFolder { Name = di.Name, Path = di.FullName });
+                var fileNfoName = String.Format("{0}\\tvshow.nfo", di.FullName);
+                if (File.Exists(fileNfoName))
+                {
+                    try
+                    {
+                        XmlDocument doc = new XmlDocument();
+                        doc.Load(File.Open(fileNfoName, FileMode.Open));
+                        XmlNode node = doc.DocumentElement.SelectSingleNode("/tvshow/id");
+                        if (node == null)
+                        {
+                            results.Add(new UnmappedFolder { Name = di.Name, Path = di.FullName });
+                        }
+                        else
+                        {
+                            String tvid = node.InnerText;
+                            results.Add(new UnmappedFolder { Name = String.Format("tvdb:{0}", tvid), Path = di.FullName });
```

Comment:
The name should still be the name, any additional information should be returned via a separate property, which also means the API and UI will need to be updated to pull in that information.